src/sort/bool.rs

```diff
@@ -0,0 +1,55 @@
+use crate::ast::Literal;
+
+use super::*;
+
+#[derive(Debug)]
+pub struct BoolSort {
+    name: Symbol,
+}
+
+impl BoolSort {
+    pub fn new(name: Symbol) -> Self {
+        Self { name }
+    }
+}
+
+impl Sort for BoolSort {
+    fn name(&self) -> Symbol {
+        self.name
+    }
+
+    fn as_arc_any(self: Arc<Self>) -> Arc<dyn Any + Send + Sync + 'static> {
+        self
+    }
+
+    #[rustfmt::skip]
+    fn register_primitives(self: Arc<Self>, eg: &mut TypeInfo) {
+        add_primitives!(eg, "not" = |a: bool| -> bool { !a });
```

Comment:
Have you considered symbol names like `!`, `&&`, and `||` to avoid conflict with potentially common names?

Reply:
Good idea, I'll do that

Reply:
I don't like these names because they are similar to `=` and `!=` and `>` which are filters